internal/transport/http2_server.go

```diff
@@ -266,6 +266,13 @@ func newHTTP2Server(conn net.Conn, config *ServerConfig) (_ ServerTransport, err
 	// Check the validity of client preface.
 	preface := make([]byte, len(clientPreface))
 	if _, err := io.ReadFull(t.conn, preface); err != nil {
+		// In deployments where a gRPC server runs behind a cloud load balancer
+		// which performs regular TCP level health checks, the connection is
+		// closed immediately by the latter. Skipping the error here will help
+		// reduce log clutter.
+		if err == io.EOF {
+			return nil, nil
+		}
 		return nil, connectionErrorf(false, err, "transport: http2Server.HandleStreams failed to receive the preface from client: %v", err)
```

Comment:
This seems to only matter for plaintext connections, or else the peer is finishing the handshake.

Is this possibly an incomplete fix?  Where do we handle handshaker errors?  Do they log?

Reply:
> My understanding is that cloud load balancers which perform TCP handshakes do not use TLS

But these logs are from the server, so it doesn't matter what the client is attempting to use.  If it connects without TLS then the server's handshaker will fail it before we get here.

Reply:
What we could do is that we could handle `io.EOF` as a special case here just like how `err != credentials.ErrConnDispatched` is being handled, and not add a log here.
https://github.com/grpc/grpc-go/blob/master/server.go#L843-L855

Reply:
Yes, something like that might be a good idea, too.

Reply:
Done. Made the change to `server.go`.